rotkehlchen/tests/unit/test_savepoints.py

```diff
@@ -0,0 +1,44 @@
+import sqlite3
+
+import pytest
+
+from rotkehlchen.globaldb.handler import GlobalDBHandler
+
+
+def test_savepoints():
+    conn = GlobalDBHandler().conn
+    conn.execute('CREATE TABLE a(b INTEGER PRIMARY KEY)')
+    with pytest.raises(sqlite3.Error):
+        with conn.savepoint_ctx('point1') as cursor1:
+            assert conn.savepoints == ['point1']
+            cursor1.execute('INSERT INTO a VALUES (1), (2)')  # insert something
+            with conn.savepoint_ctx('point2') as cursor2:
+                assert conn.savepoints == ['point1', 'point2']
+                # insert more and check multiple nested savepoints
+                cursor2.execute('INSERT INTO a VALUES (3)')
+                with pytest.raises(sqlite3.Error):
+                    # Also check savepoints with the same name. In this case, according to sqlite's
+                    # docs, only the most recent savepoint should be released/rolled back.
+                    with conn.savepoint_ctx('point2') as cursor3:
+                        assert conn.savepoints == ['point1', 'point2', 'point2']
+                        cursor3.execute('INSERT INTO a VALUES (4)')
+                        cursor3.execute('INSERT INTO a VALUES (1)')  # make an error
+                # check that 4 was removed
+                assert cursor2.execute('SELECT 4 IN (SELECT b FROM a)').fetchone()[0] == 0
+                assert conn.savepoints == ['point1', 'point2']
+            # Here we are at the moment when "point2" is released
+            # check that 3 exists
+            assert cursor1.execute('SELECT 3 IN (SELECT b FROM a)').fetchone()[0] == 1
+            assert conn.savepoints == ['point1']
+            cursor4 = conn.enter_savepoint('point4')  # test manual enter
+            assert conn.savepoints == ['point1', 'point4']
+            cursor4.execute('DELETE FROM a WHERE b=1')
+            conn.release_savepoint('point4')  # test manual release
+            assert conn.savepoints == ['point1']
+            # check that 1 was removed
+            assert cursor1.execute('SELECT 1 IN (SELECT b FROM a)').fetchone()[0] == 0
+            # Finally break point1 and check that all insertions/deletions were undone
+            cursor1.execute('INSERT INTO a VALUES (2)')  # 2 already exists
+
+    assert len(conn.savepoints) == 0
```

Comment:
generally adjust the tests for what I told you to implement in the db driver part of the review. And keep 3 tests. One with named savepoints, one without giving specific names, and one with having named one at start, then go deeper with no names but skip directly back to the named one. 

Guess the last test can be combined with one of the other 2